community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStores.java

```diff
@@ -142,9 +142,24 @@ public boolean test( StoreType type )
 
         verifyRecordFormat();
         stores = new Object[StoreType.values().length];
-        for ( StoreType type : storeTypes )
+        try
+        {
+            for ( StoreType type : storeTypes )
+            {
+                getOrCreateStore( type );
+            }
+        }
+        catch ( Exception initException )
```

Comment:
`RuntimeException`

Reply:
can IOUtils.closeAll help here?

Reply:
It's a slightly different situation than what IOUtil.closeAll is good for. We're only closing one thing, kind of, and only if the initialisation fails.